warp/src/raygun/mod.rs

```diff
@@ -386,7 +386,7 @@ pub enum ConversationType {
     #[display(fmt = "direct")]
     Direct,
     #[display(fmt = "group")]
-    Group { open: bool },
+    Group { settings: GroupSettings },
```

Comment:
If we are to go with this "setting" then it might be best to leave this as `Group` since the conversation type does not intend to store any additional metadata like `GroupSettings` and instead just have this apart of the `ConversationDocument` as a separate field and have a new trait function to obtain the struct from the document.

Reply:
> since the conversation type does not intend to store any additional metadata like `GroupSettings`

That might not be the original intention but since it's group-specific, it's best be only available/accessible through this enum (if kept in `Conversation`, read below).

> instead just have this apart of the `ConversationDocument` as a separate field and have a new trait function to obtain the struct from the document.

So only `ConversationDocument`and not `Conversation`? 🤔

Reply:
We could have another type:

```rust
enum ConversationSettings {
   Direct(DirectionConversationSettings),
   Group(GroupSettings),
}
```

And use that as the direct field but that's really just duplicating `ConversationType` enum.

Reply:
> That might not be the original intention but since it's group-specific, it's best be only available/accessible through this enum (if kept in `Conversation`, read below).

I dont mind it being apart of a separate function to obtain the `GroupSettings` since storing it through this enum would not really make sense, especially if `GroupSettings` is ever expanded. 

> So only `ConversationDocument`and not `Conversation`? 🤔

Correct, since `ConversationDocument` that is internal, that could be expanded more if needed.

Reply:
I made the changes as we agreed and for now I added settings to `Conversation`as well (because of the conversion to `ConversationDocument` but we can change that later.

However, the group tests now fail because somehow the conversation ends up with `ConversationSettings::Direct`:

```rust
thread 'test::create_empty_group_conversation' panicked at extensions/warp-ipfs/tests/group.rs:42:9:
assertion `left == right` failed
  left: Direct(DirectConversationSettings)
 right: Group(GroupSettings { members_can_add_participants: false })
```

Need to figure out what I missed.